checker/src/org/checkerframework/checker/samelen/SameLenTransfer.java

```diff
@@ -0,0 +1,134 @@
+package org.checkerframework.checker.samelen;
+
+import javax.lang.model.element.AnnotationMirror;
+import javax.lang.model.type.TypeKind;
+import org.checkerframework.dataflow.analysis.FlowExpressions;
+import org.checkerframework.dataflow.analysis.FlowExpressions.Receiver;
+import org.checkerframework.dataflow.analysis.TransferInput;
+import org.checkerframework.dataflow.analysis.TransferResult;
+import org.checkerframework.dataflow.cfg.node.*;
+import org.checkerframework.framework.flow.CFAnalysis;
+import org.checkerframework.framework.flow.CFStore;
+import org.checkerframework.framework.flow.CFTransfer;
+import org.checkerframework.framework.flow.CFValue;
+
+/**
+ * The transfer function for the SameLen checker. Contains two interesting cases: * If an array is
+ * created using another array's length for its length, then the arrays have the same length. * If
+ * the lengths of two arrays are explicitly checked to be equal, they have the same length.
+ */
+public class SameLenTransfer extends CFTransfer {
+
+    // The ATF (Annotated Type Factory).
+    private SameLenAnnotatedTypeFactory aTypeFactory;
+
+    /** Easy shorthand for SameLenUnknown.class, basically. */
+    public static AnnotationMirror UNKNOWN;
+
+    private CFAnalysis analysis;
+
+    public SameLenTransfer(CFAnalysis analysis) {
+        super(analysis);
+        this.analysis = analysis;
+        aTypeFactory = (SameLenAnnotatedTypeFactory) analysis.getTypeFactory();
+        UNKNOWN = SameLenAnnotatedTypeFactory.UNKNOWN;
+    }
+
+    @Override
+    public TransferResult<CFValue, CFStore> visitAssignment(
+            AssignmentNode node, TransferInput<CFValue, CFStore> in) {
+        TransferResult<CFValue, CFStore> result = super.visitAssignment(node, in);
+
+        // Check if an array is being created.
+
+        if (node.getExpression() instanceof ArrayCreationNode) {
+            ArrayCreationNode acNode = (ArrayCreationNode) node.getExpression();
+            if (acNode.getDimensions().size() == 1 && isArrayLengthAccess(acNode.getDimension(0))) {
+                // This array that's being created is the same size as the other one.
+                AnnotationMirror combinedSameLen =
+                        aTypeFactory.createCombinedSameLen(
```

Comment:
This is impossible to read.  Please extract some of the arguments to local variables.